azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -937,6 +942,14 @@ def _run_op(self, op):
                         self.name, op.name, self.state
                     )
                 )
+                self.state = ReconnectState.LOGICALLY_CONNECTED
+                # We don't send this op down.  Instead, we send a new connect op down.  This way,
+                # we can distinguish between connect ops that we're handling (they go into the
+                # queue) and connect ops that we are sending down.
```

Comment:
I wonder if this indicates perhaps there should be a concrete separate op so this isn't just by convention. Something to consider going forth